common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -192,11 +196,7 @@ public void streamSent(String streamId) {
 
   @Override
   public long chunksBeingTransferred() {
-    long sum = 0L;
-    for (StreamState streamState: streams.values()) {
-      sum += streamState.chunksBeingTransferred.get();
-    }
```

Comment:
Does this becomes a blocker? Or how much performance improvement we can gain in the new approach?

Reply:
It makes no big different if the shuffle server handles not many chunks. While in our production environment, we found when the number of chunks reach 100,000 or more, sometimes most of the cpu resource are occupied by iterating and calculation the total number. Then no cpu resource to handle request and response data, which makes everything stuck.

Reply:
Have you measured how long the thread is being stuck on calculation here? And how frequently this method is called?

It would be ideal to craft a simple benchmark code and experiment with some variations: num of chunks, contentions (no contention, one thread calling chunkSent, N threads doing the same).

Reply:
Looks like the method is not called at only once, so the optimization seems to make sense. Still would be better to have some numbers to back up rationalization of the patch.